lib/OpenQA/Scheduler/Scheduler.pm

```diff
@@ -1179,23 +1179,39 @@ sub job_schedule_iso {
         asset_register(%$a);
     }
     my $noobsolete = delete $args{_NOOBSOLETEBUILD};
-    # ISOURL == ISO download. If the ISO already exists, skip the
-    # download step (below) entirely by leaving $isodlpath unset.
-    my $isodlpath;
-    if ($args{ISOURL}) {
+    # Any arg name ending in _URL is special: it tells us to download
+    # the file at that URL before running the job
+    my %downloads = ();
+    for my $arg (keys %args) {
+        next unless ($arg =~ /_URL$/);
```

Comment:
this hit on a rather innocent URL in our tests - is there a way to only trigger on URLs you would also download?